src/hotspot/share/opto/superword.cpp

```diff
@@ -1996,291 +1961,20 @@ void PackSet::verify() const {
-  bool schedule_success() const { return _schedule_success; }
-
-  // Create nodes (from packs and scalar-nodes), and add edges, based on the dependency graph.
-  void build() {
-    const PackSet& packset = _slp->packset();
-    const GrowableArray<Node*>& body = _slp->body();
-    // Map nodes in packsets
-    for (int i = 0; i < packset.length(); i++) {
-      Node_List* p = packset.at(i);
-      int pid = new_pid();
-      for (uint k = 0; k < p->size(); k++) {
-        Node* n = p->at(k);
-        set_pid(n, pid);
-        assert(packset.get_pack(n) == p, "matching packset");
-      }
-    }
-
-    int max_pid_packset = _max_pid;
-
-    // Map nodes not in packset
-    for (int i = 0; i < body.length(); i++) {
-      Node* n = body.at(i);
-      if (n->is_Phi() || n->is_CFG()) {
-        continue; // ignore control flow
-      }
-      int pid = get_pid_or_zero(n);
-      if (pid == 0) {
-        pid = new_pid();
-        set_pid(n, pid);
-        assert(packset.get_pack(n) == nullptr, "no packset");
-      }
-    }
-
-    // Map edges for packset nodes
-    VectorSet set;
-    for (int i = 0; i < packset.length(); i++) {
-      Node_List* p = packset.at(i);
-      set.clear();
-      int pid = get_pid(p->at(0));
-      for (uint k = 0; k < p->size(); k++) {
-        Node* n = p->at(k);
-        assert(pid == get_pid(n), "all nodes in pack have same pid");
-        for (VLoopDependencyGraph::PredsIterator preds(_slp->dependency_graph(), n); !preds.done(); preds.next()) {
-          Node* pred = preds.current();
-          int pred_pid = get_pid_or_zero(pred);
-          if (pred_pid == pid && _slp->is_marked_reduction(n)) {
-            continue; // reduction -> self-cycle is not a cyclic dependency
-          }
-          // Only add edges once, and only for mapped nodes (in body)
-          if (pred_pid > 0 && !set.test_set(pred_pid)) {
-            incnt_set(pid, incnt(pid) + 1); // increment
-            out(pred_pid).push(pid);
-          }
-        }
-      }
-    }
-
-    // Map edges for nodes not in packset
-    for (int i = 0; i < body.length(); i++) {
-      Node* n = body.at(i);
-      int pid = get_pid_or_zero(n); // zero for Phi or CFG
-      if (pid <= max_pid_packset) {
-        continue; // Only scalar-nodes
-      }
-      for (VLoopDependencyGraph::PredsIterator preds(_slp->dependency_graph(), n); !preds.done(); preds.next()) {
-        Node* pred = preds.current();
-        int pred_pid = get_pid_or_zero(pred);
-        // Only add edges for mapped nodes (in body)
-        if (pred_pid > 0) {
-          incnt_set(pid, incnt(pid) + 1); // increment
-          out(pred_pid).push(pid);
-        }
-      }
-    }
-  }
-
-  // Schedule nodes of PacksetGraph to worklist, using topsort: schedule a node
-  // that has zero incnt. If a PacksetGraph node corresponds to memops, then add
-  // those to the memops_schedule. At the end, we return the memops_schedule, and
-  // note if topsort was successful.
-  Node_List schedule() {
```

Comment:
Note: replaced with `VTransformGraph::schedule`. Instead of topsort I now do reverse-post-order.